src/mongo/client/dbclient_rs.h

```diff
@@ -203,6 +203,22 @@ class MONGO_CLIENT_API DBClientReplicaSet : public DBClientBase {
      */
     static void setAuthPooledSecondaryConn(bool setting);
 
+    virtual int getMaxWireVersion() {
+        if (!_master.get()) {
+            return checkMaster()->getMaxWireVersion();
```

Comment:
can we always call checkMaster() ? so the entire body would be `return checkMaster()->getMaxWireVersion();`. If there hasn't been a failover checkMaster() is cheap - if it isn't then we need to call it anyway.